File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here

  let newStr = "";

  for (let i = str.length - 1; i >= 0; i--) {
    newStr = newStr + str[i];
  }

  return newStr;
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file

/// I want to return reversed input
/// Create empty string to store iterated letter
/// Use for loop to iterate from last length until 0
/// Store each letter in var
/// Return

/// THIRD TIME

//// First create variable to store iterated letter
//// Iterate each letter from last length until 0
//// Store iterated letter to above var

// let newStr = ""

// for (let i = 0; i < str.length ; i++) {
//   newStr += str[i]
// }

// return newStr

//// Second Time /////
// I want to console log reversed word using iteration
// First, create empty variable to store iterated letter
// Second, iterate word from last length then add to above variable
// Then, return variable

// let newStr = [];

// for (let i = str.length-1; i >= 0; i--) {
//   newStr += str[i];
// }

// return newStr;

//// FIrst Time /////
//// I want to input reversed letters 1 by 1
//// First, create empty variable to store iterated letters
//// Iterate from last length-1, until equal to 0, decrement
//// Add iterated letters to variable
//// Then, return variable

// function reverseString(str) {
//   // type your code here

//   let newString = "";

//   for (let i = str.length - 1; i >= 0; i--) {
//     newString += str[i];
//   }

//   return newString;

//   // let newStr = str.split("").reverse().join("");

//   // return newStr
// }

File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here

  newStr = ""
  i = 0

  while i < str.length
    newStr = str[i] + newStr
    i += 1
  end

  newStr

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file



#### Third Attempt

# newStr = ""

# i = 0
# length = str.length

# length.times do
#   newStr = newStr + str[i]
#   i += 1
# end

# newStr



#### Second Attempt
### I want to iterate string one by one until the last length
### Then, add iterated letter to variable

# newStr = ""
# i = str.length

# while i > 0 do
#   i -= 1
#   newStr = newStr + str[i]
# end

# newStr



######## First Attempt
## I want to input reveresed string

## First, create variable and split string 
## Then, iterate each letter AND ADD LETTER IN FRONT OF PREVIOUS LETTER


  ### Using while method

  # Varilable to store iterated letter
  # newStr = ""

  # While i is less than string length (5), in other words,
  # while i is less than 5, add iterated letter in front of exsiting letter
  # increment i
  # i = 0
  # while i < str.length
  #   newStr = str[i] + newStr
  #   i += 1
  # end

  # return newStr

  

  ### Using .each method

  ## Split str and create empty variable to store letters
  # newStr = str.split("")
  # reversedStr = []

  ## Iterate each letter and add in front of previous letter
  # newStr.each do |s| 
  #   reversedStr.unshift(s)
  # end

  # return reversedStr.join("")

  ### Using .reverse method
  # newStr = str.split("").reverse().join("")
  # return newStr
 